tests/test_cli_get_set.py

```diff
@@ -63,53 +63,54 @@ def test_set_scalar(self):
         self._set(['ethernets.eth0.dhcp4=true'])
         self.assertTrue(os.path.isfile(self.path))
         with open(self.path, 'r') as f:
-            self.assertIn('network:\n  ethernets:\n    eth0:\n      dhcp4: true', f.read())
+            self.assertIs(True, yaml.safe_load(f)['network']['ethernets']['eth0']['dhcp4'])
 
     def test_set_scalar2(self):
         self._set(['ethernets.eth0.dhcp4="yes"'])
         self.assertTrue(os.path.isfile(self.path))
         with open(self.path, 'r') as f:
-            self.assertIn('network:\n  ethernets:\n    eth0:\n      dhcp4: \'yes\'', f.read())
+            # XXX: the previous version using PyYAML would keep the "yes" variant but the round-trip
+            # through libnetplan doesn't keep formatting choices (yes is a keyword same as true)
+            self.assertIs(True, yaml.safe_load(f)['network']['ethernets']['eth0']['dhcp4'])
```

Comment:
thought (non-blocking): Could we somehow make use of the new "dirty" tracking functionality to keep formatting choices?

Reply:
I suppose we could, by storing the actual scalar data as the value in the hashtable. I'm not sure it's worth the hassle, though.

Reply:
ACK. we could still do this in the future, if needed. Let's keep it out of this (already big) PR.